utils/image.go

```diff
@@ -36,9 +63,14 @@ func ExtractOCIImage(targetImage, targetDestination, targetPlatform string, isLo
 		return err
 	}
 
-	img, err := getimage(ref, *platform, isLocal)
+	// Try local then remote, like the docker client does
+	// This should be pretty fast
+	img, err = getimage(ref, *platform, true)
```

Comment:
I was about to say that we'd better have 2 functions instead of passing boolean arguments (e.g. `getLocalImage` and `getRemoteImage`) but the realized that in the case of local, it's an one-liner: `daemon.Image(ref)`.

There is no reason to complicate `getimage` signature like this. Just do a check here:

```
image, err := daemon.Image(ref)
if err != nil {
  img, err = getimage(ref, *platform)
  if err != nil {
    return err
  }
}
```

(non-tested code)

Reply:
Yeah, but the idea is to expand the underlying method of getting the stuff, i.e. abstract it so we can add stuff in there, like the retry transport (which can also be applied to the local daemon btw, it just doesnt make sense in a local environment...I think? Not sure if the local daemon can also benefit from the retry, maybe on corner cases) and others (maybe we need to also do auth in the local daemon in some cases?)

So I do expect the getimage to expand in the future with extra things, while the facing `ExtractOCIImage` seems more abstract. parse, get the image, extract, done

Reply:
Methods/Functions should be doing only one thing. Boolean arguments mean they don't. But anyway, I don't want to be pedantic, keep it as it if you prefer.